source/extensions/omni.isaac.lab/omni/isaac/lab/assets/articulation/articulation.py

```diff
@@ -539,6 +539,8 @@ def write_joint_limits_to_sim(
             env_ids = env_ids[:, None]
         # set into internal buffers
         self._data.joint_limits[env_ids, joint_ids] = limits
+        # update default joint pos to stay within the new limits
+        self._data.default_joint_pos = torch.clamp(self._data.default_joint_pos, limits[..., 0], limits[..., 1])
```

Comment:
Assuming users don't set the joint limits often, I think it would be good to throw a warning about this clipping if it affects the default. What do you think?

Reply:
sure, added a warning message